gradle/wrapper/gradle-wrapper.properties

```
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.1-rc-1-bin.zip
```

To stabilize the build, we can keep previous config:

```suggestion
distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.3-bin.zip
```

and use `.travis.yml` to enable build with Gradle 5.1 RC1.

---

In `.travis.yml`, you can add one build matrix item like:

```yml
env:
  matrix:
    - GRADLE_VERSION=default
    - GRADLE_VERSION=5.0
    - GRADLE_VERSION=5.1-rc-1
```

---

I don't think this change is backwards compatible, because the API changed, so it would break on Gradle 5.0 or less. This is because `CollectionCallbackActionDecorator` is new in Gradle 5.1 https://github.com/gradle/gradle/commits/master/subprojects/core/src/main/java/org/gradle/api/internal/CollectionCallbackActionDecorator.java

I only kept `    - GRADLE_VERSION=5.1-rc-1` in the matrix for now.

to make this backwards compatible, it would require probably a rewrite on how things are done in the plugin, but I don't think there is a clear guideline from Gradle team on this.

---

OK then I believe that what we should do is not merging this PR but finding the way to stop depending on internal API like this `CollectionCallbackActionDecorator`.